### PR TITLE
Add InternalDb unit tests and fix verse-255 boundary bug (REM-18c)

### PR DESCRIPTION
**Pull request by yukuku**
## Summary

Adds **39 unit tests** for `InternalDb` covering the risk areas called out by **step 18c** of the tech-debt remediation plan: marker CRUD, label ordering, highlight storage, and attribute loading — plus a fix for a latent boundary bug surfaced by review.

## Why

`InternalDb` is one of the most error-prone surfaces in the app — it backs bookmarks, notes, highlights, labels, sync, and the verse-attribute overlay, but has had zero unit-test coverage. The existing `InternalDbHelper.onUpgrade` comments reference past migration bugs that shipped unnoticed. Step 18c (alongside 18a Highlights and 18b Sync) is the last missing piece of cross-cutting test coverage for the core storage layer.

## Bug fix: verse-255 boundary

`InternalDb.countMarkersForBookChapter` and `InternalDb.putAttributes` computed `ariMax = ari_bookchapter | 0xff` and then queried `ari >= ariMin AND ari < ariMax`, which silently dropped any marker on verse 255 (whose ari equals `ariMax`). A third method on the same class — `getHighlightColorRgb(int, IntArrayList)` — already used the inclusive bound `ari <= ariMax`, making the three range queries inconsistent. This PR flips the bounds in the first two methods to `<=` to match.

No Bible chapter has 255 verses (Psalm 119, the longest, has 176), so no real data was ever affected — but the boundary should be correct for consistency, and the fix is one character per SQL statement. Two dedicated verse-255 regression tests lock the fix down.

## Test strategy

Robolectric is used because `InternalDbHelper` extends Android's `SQLiteOpenHelper` — cursors, transactions, and `getReadableDatabase()` / `getWritableDatabase()` cannot run under plain JUnit.

The test class uses `@Config(application = Application::class, sdk = [34])` so the real `yuku.alkitab.base.App.onCreate` path (Firebase / PRDownloader / FCM / extension receivers) does **not** run. It reuses the test-scope shadows of `android.util.Log` and `com.google.firebase.crashlytics.FirebaseCrashlytics` that were added for REM-18a/b, so `AppLog`'s static initializer loads without bootstrapping `FirebaseApp`. `yuku.afw.App.context` is wired manually in `@Before`. Because no `sync_simpleToken` preference is set, `Sync.notifySyncNeeded` early-returns — DB mutations under test don't trigger any network or background work.

`sdk = [34]` matches the project's existing Robolectric test (`VerseActionModeControllerTest` from REM-07); Robolectric 4.13 (the pinned version) does not support API 35. Bumping the Robolectric version is out of scope for this PR.

Test function names follow the Kotlin backtick-sentence convention documented in CLAUDE.md.

## Coverage

- **Marker CRUD** — `insertMarker`, `insertOrUpdateMarker`, `getMarkerById/Gid`, `listMarkersForAriKind` (modifyTime desc), `listAllMarkers`, `deleteMarkerById` cascade into `Marker_Label`, `countMarkersForBookChapter` (including verse-255 regression)
- **Label ordering** — `insertLabel` / `getLabelMaxOrdering` increments, `reorderLabels` up and down shift logic, `sortLabelsAlphabetically` (case-insensitive), `listLabelsByMarker` ordering
- **Highlight storage** — `updateOrInsertHighlights` insert/update/delete, `updateOrInsertPartialHighlight` including dedup of sync-duplicates, `getHighlightColorRgb` single and multi-verse (uniform / mixed / none)
- **Attribute loading** — `putAttributes` bookmarks, notes, multi-verse highlight spread across `verseCount`, latest-wins on duplicate ari, book-chapter filtering (including verse-255 regression)
- **Marker_Label helpers** — `insertOrUpdateMarker_Label`

## Other production-code concerns flagged for follow-up

Per the CLAUDE.md guideline to assume production code is correct and flag (not work around) anything suspicious, these were noticed while writing tests but are **not fixed in this PR**:

1. **`InternalDb.deleteMarkerById`** (`InternalDb.java:171-184`) dereferences `marker.gid` after `getMarkerById(_id)`. If the id doesn't exist, `getMarkerById` returns `null` and the next line NPEs before `db.delete` runs. Benign in practice (callers always pass known-good ids from the UI) but a real latent bug.
2. **`InternalDb.deleteLabelAndMarker_LabelsByLabelId`** (`InternalDb.java:826-832`) has the same null-deref shape: `getLabelById(_id)` can return `null`, then `label.gid` NPEs.
3. **`InternalDb.countMarkersForBookChapter`** (`InternalDb.java:240-247`) caches a `SQLiteStatement` tied to the database handle from the first call. If the DB is ever closed/reopened, the cached statement would be stale. In practice the helper lives for the app lifetime, so it doesn't bite.

All three are one-line fixes suitable for a separate PR.

## Test plan

- [x] `./gradlew :Alkitab:testPlainDebugUnitTest` — 213 tests pass (39 new + 174 pre-existing), 0 failures
- [x] `./gradlew :Alkitab:testPlainReleaseUnitTest` — 213 tests pass, 0 failures
- [ ] CI green

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+ * so we don't need to bootstrap FirebaseApp here either.
+ *
+ * [yuku.afw.App.context] is wired manually in [setUp] so indirect dependencies
+ * (Gson singleton via `App.getDefaultGson()`, `Preferences`) can resolve.
+ * Because no `sync_simpleToken` preference is set,
+ * [yuku.alkitab.base.sync.Sync.notifySyncNeeded] early-returns, so the DB
+ * mutations under test don't trigger any network or background work.
+ */
+@RunWith(RobolectricTestRunner::class)
+@Config(application = Application::class, sdk = [34])
```
[the remark this answers is not in the document]
Comment (yukuku) [reply Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt:47]: Good catch — I initially bumped this to 35 as you suggested, but Robolectric 4.13 (the version pinned in `build.gradle`, added by REM-07) does not yet support API 35. The existing Robolectric test on this branch (`VerseActionModeControllerTest` from REM-07) also uses `sdk = [34]`, so I've kept the value at 34 to match project convention and the available Robolectric.

Bumping `robolectricVersion` to 4.14.x (the first release with API 35 support) would enable `sdk = [35]` here, but it would also re-run all existing Robolectric tests against the new version — that's out of scope for a test-only PR. Happy to do the bump as a separate follow-up if you'd like.

File: Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+        assertEquals(1, db.listMarker_LabelsByMarker(marker).size)
+
+        db.deleteMarkerById(marker._id)
+
+        assertNull(db.getMarkerById(marker._id))
+        assertEquals(0, db.listMarker_LabelsByMarker(marker).size)
+    }
+
+    @Test
+    fun `countMarkersForBookChapter counts only markers whose ari falls inside the book-chapter range`() {
```
[the remark this answers is not in the document]
Comment (yukuku) [reply Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt:194]: Good catch, and you're right — this was one of the concerns I flagged in the PR description (item #3 of the production-code concerns). I added a dedicated test in this revision that documents the current behavior:

```kotlin
@Test
fun `countMarkersForBookChapter currently excludes verse 255 because its upper bound is exclusive (documents latent bug)`() {
    // …
    db.insertMarker(Ari.encode(0, 1, 254), Marker.Kind.bookmark, "a", 1, now, now)
    db.insertMarker(Ari.encode(0, 1, 255), Marker.Kind.bookmark, "b", 1, now, now)

    // verse 254 is counted, verse 255 is not — that's the latent bug.
    assertEquals(1, db.countMarkersForBookChapter(Ari.encode(0, 1, 0)))
}
```

Per the CLAUDE.md guideline ("assume production code is correct… flag obvious bugs rather than silently work around them"), the test pins down the buggy behavior rather than patching `InternalDb.java`. In practice no Bible chapter has 255 verses (Psalm 119, the longest, has 176), so this boundary never fires against real data — which is probably why it's gone unnoticed. The comment on the test and the amended commit message both call out the inconsistency with `getHighlightColorRgb`'s inclusive bound so it's on the record for a follow-up fix.

File: Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+        val verses = IntArrayList().apply { add(1); add(2) }
+        assertEquals(-1, db.getHighlightColorRgb(ariBc, verses))
+    }
+
+    // endregion
+
+    // region Attribute loading
+
+    @Test
+    fun `putAttributes fills the bookmark and note count arrays indexed by verse-1`() {
```
[the remark this answers is not in the document]
Comment (yukuku) [reply Alkitab/src/test/java/yuku/alkitab/base/storage/InternalDbTest.kt:541]: Agreed. I added a companion test for `putAttributes` in this revision that asserts the same behavior:

```kotlin
@Test
fun `putAttributes currently excludes verse 255 because its upper bound is exclusive (documents latent bug)`() {
    // …
    db.insertMarker(Ari.encode(0, 1, 254), Marker.Kind.bookmark, "v254", 1, now, now)
    db.insertMarker(Ari.encode(0, 1, 255), Marker.Kind.bookmark, "v255", 1, now, now)
    // …
    assertEquals(1, bookmarkCounts[253]) // verse 254 is counted
    assertEquals(0, bookmarkCounts[254]) // verse 255 is dropped — that's the latent bug
}
```

Like the `countMarkersForBookChapter` companion above, the test documents the current behavior rather than changing `InternalDb.java`. Both tests also note that `getHighlightColorRgb` uses the inclusive `ari <= ariMax`, so the three methods are inconsistent — worth a follow-up PR to reconcile.
